### DISCUSSION TITLE
problem in using third party libraries in ssr mode

### DISCUSSION THREAD

**Discussion by zeynab-rs** [quasarframework/quasar#11287] [CLI - SSR mode]
I use third party libraries like swiper.js and particles.js in quasar v2 as the same as in quasar v1. But i get error for them in ssr mode in v2. How to handle that?

**Comment by MilosPaunovic** [comment]
Quasar `v2` is using `Vue3`, unlike `v1` which is on `Vue2`. First step is to ensure that libraries you're using support `Vue3`.

It's always better to provide some kind of code reproduction so you could get better support.

**Reply by zeynab-rs** [reply]
Problem is with SSR mode because of for example using window object in particles.js codes. In SPA mode, everything work fine.

**Comment by hawkeye64** [comment]
@heyran if you have boot files for them, you can register the boot file in a way that they are not run on the server: https://quasar.dev/quasar-cli/developing-ssr/writing-universal-code#boot-files
